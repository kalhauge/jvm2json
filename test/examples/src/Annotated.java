import java.lang.annotation.*;

public class Annotated {

  @Retention(RetentionPolicy.CLASS)
  @Target(value=ElementType.TYPE_USE)
  public static @interface Ann {}

  public void test () { 
    @Ann Integer a = 0;
  }
}
