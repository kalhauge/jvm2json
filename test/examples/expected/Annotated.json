{
  "name": "Annotated",
  "access": [
    "public",
    "super"
  ],
  "typeparams": [],
  "super": {
    "name": "java/lang/Object",
    "inner": null,
    "args": [],
    "annotations": []
  },
  "interfaces": [],
  "fields": [],
  "methods": [
    {
      "name": "<init>",
      "access": [
        "public"
      ],
      "typeparams": [],
      "params": [],
      "returns": {
        "type": null,
        "annotations": []
      },
      "code": {
        "max_stack": 1,
        "max_locals": 1,
        "exceptions": [],
        "stack_map": null,
        "lines": [
          {
            "offset": 0,
            "line": 3
          }
        ],
        "annotations": [],
        "bytecode": [
          {
            "offset": 0,
            "opr": "load",
            "type": "ref",
            "index": 0
          },
          {
            "offset": 1,
            "opr": "invoke",
            "access": "special",
            "method": {
              "is_interface": false,
              "ref": {
                "kind": "class",
                "name": "java/lang/Object"
              },
              "name": "<init>",
              "args": [],
              "returns": null
            }
          },
          {
            "offset": 4,
            "opr": "return",
            "type": null
          }
        ]
      },
      "annotations": [],
      "exceptions": [],
      "default": null
    },
    {
      "name": "test",
      "access": [
        "public"
      ],
      "typeparams": [],
      "params": [],
      "returns": {
        "type": null,
        "annotations": []
      },
      "code": {
        "max_stack": 1,
        "max_locals": 2,
        "exceptions": [],
        "stack_map": null,
        "lines": [
          {
            "offset": 0,
            "line": 10
          },
          {
            "offset": 3,
            "line": 11
          }
        ],
        "annotations": [
          {
            "target": {
              "type": "LocalVariableDeclaration",
              "bounds": [
                {
                  "start": 3,
                  "end": 4,
                  "localindex": 1
                }
              ]
            },
            "path": [],
            "annotation": {
              "type": "Annotated$Ann",
              "is_runtime_visible": false,
              "values": {}
            }
          }
        ],
        "bytecode": [
          {
            "offset": 0,
            "opr": "push",
            "value": {
              "type": "integer",
              "value": 0
            }
          },
          {
            "offset": 1,
            "opr": "invoke",
            "access": "static",
            "method": {
              "is_interface": false,
              "ref": {
                "kind": "class",
                "name": "java/lang/Integer"
              },
              "name": "valueOf",
              "args": [
                "int"
              ],
              "returns": {
                "kind": "class",
                "name": "java/lang/Integer"
              }
            }
          },
          {
            "offset": 4,
            "opr": "store",
            "type": "ref",
            "index": 1
          },
          {
            "offset": 5,
            "opr": "return",
            "type": null
          }
        ]
      },
      "annotations": [],
      "exceptions": [],
      "default": null
    }
  ],
  "bootstrapmethods": [],
  "enclosingmethod": null,
  "innerclasses": [
    {
      "class": "Annotated$Ann",
      "outer": "Annotated",
      "name": "Ann",
      "access": [
        "public",
        "static",
        "interface",
        "abstract",
        "annotation"
      ]
    }
  ],
  "annotations": [],
  "version": [
    63,
    0
  ]
}
